Update URLs based on redirects

davisonio opened the pull request:
This changes some of the URLs to use their new location.

I also experienced a few sites which 404'd, see #1742

vhf commented on free-programming-books-hu.md:
```diff
@@ -22,7 +22,7 @@
 * [Ember, információ, rendszer](http://mek.oszk.hu/11100/11122/index.phtml) - Halassy Béla (Word, PDF)
 * [Formális nyelvek](http://mek.oszk.hu/05000/05099/index.phtml) - Bach Iván (PDF)
-* [Kanban és Scrum mindkettőből a legjobbat](http://www.adaptiveconsulting.hu/dokumentumok) - Henrik Kniberg, Mattias Skarin, ford.: Csutorás Zoltán és Marhefka István (PDF)
+* [Kanban és Scrum mindkettőből a legjobbat](http://www.adaptiveconsulting.hu) - Henrik Kniberg, Mattias Skarin, ford.: Csutorás Zoltán és Marhefka István (PDF)
```
This book is not free, we should get rid of it.

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/9d81cbc6bf6251935fd3f88df59ae66843e2f42e

vhf commented on free-programming-books-it.md:
```diff
@@ -35,7 +35,7 @@
 ### Android
 
-* [Guida programmazione Android 4.2](http://www.sprik.it/guida/Android4_2.pdf) (PDF)
+* [Guida programmazione Android 4.2](http://www.sprik.it/download/guida-programmazione-android-4-2/) (PDF)
```
```diff
+* [Guida programmazione Android 4.2](http://www.sprik.it/download/guida-programmazione-android-4-2/) (PDF)
-* [Guida programmazione Android 4.2](http://www.sprik.it/download/guida-programmazione-android-4-2/)
```

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/0eb1824048451f69120ec2d32bf5f5143aaa6825

vhf commented on free-programming-books-it.md:
```diff
@@ -56,7 +56,7 @@
 ### C Sharp
 
-* [AB..C# - Guida alla programmazione](http://www.youblisher.com/files/publications/4/21542/pdf.pdf) (PDF)
+* [AB..C# - Guida alla programmazione](http://www.youblisher.com/p/21542-ABC-Guida-alla-programmazione/) (PDF)
```
```diff
* [AB..C# - Guida alla programmazione](http://www.youblisher.com/p/21542-ABC-Guida-alla-programmazione/) (PDF)
+* [AB..C# - Guida alla programmazione](http://www.youblisher.com/p/21542-ABC-Guida-alla-programmazione/)
```

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/d8a2b0b12917ec37db802ecbcd7647e77f01380f

vhf commented on free-programming-books-ja.md:
```diff
@@ -228,7 +228,7 @@
 * [ロベールのＣ＋＋教室](http://www7b.biglobe.ne.jp/~robe/cpphtml/) - ロベール
 * [cpprefjp - C++ Reference Site in Japanese](http://cpprefjp.github.io)
-* [C++11の文法と機能(C++11: Syntax and Feature)](https://ezoeryou.github.com/cpp-book/C++11-Syntax-and-Feature.xhtml) - 江添亮
+* [C++11の文法と機能(C++11: Syntax and Feature)](http://ezoeryou.github.io/cpp-book/C++11-Syntax-and-Feature.xhtml) - 江添亮
```
https here please

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/b3918b4b9aab2df3be45c419398722bd6c98ae6f

vhf commented on free-programming-books-ja.md:
```diff
@@ -296,13 +296,13 @@
 
 * [Gradle 日本語ドキュメント](http://gradle.monochromeroad.com/docs/) - Hayashi Masatoshi, Sekiya Kazuchika, Sue Nobuhiro, Mochida Shinya ([PDF](http://gradle.monochromeroad.com/docs/userguide/userguide.pdf))
-* [ビルドツールGradleスタートアップガイドの紹介](http://www.ntts.co.jp/publish/column/tec/java_03/) - 鈴木雅貴
+* [ビルドツールGradleスタートアップガイドの紹介](https://www.ntts.co.jp/column/tec/java_03/index2.html) - 鈴木雅貴
```
Could you please remove `index2.html`?

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/d2db81ac4b781ff889ab7fc333afd05b53f91b41

vhf commented on free-programming-books-ja.md:
```diff
@@ -296,13 +296,13 @@
 
 * [Grailsフレームワーク 日本語リファレンス](http://grails.jp/doc/latest/) - T.Yamamoto & Japanese Grails Doc Translating Team. Special thanks to NTT Software.
-* [Grailsを用いたファイルアップローダ作成](http://www.ntts.co.jp/publish/column/tec/java_02/) - 須藤悠
+* [Grailsを用いたファイルアップローダ作成](https://www.ntts.co.jp/column/tec/java_02/index2.html) - 須藤悠
```
Could you please remove `index2.html`?

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/d2db81ac4b781ff889ab7fc333afd05b53f91b41

vhf commented on free-programming-books-ko.md:
```diff
@@ -20,7 +20,7 @@
 
 * [Pro Git 한글 번역](http://git-scm.com/book/ko/)
-* [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown)
+* [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown/)
```
Let's remove this link. It's not in Korean.

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/68ce19c181906803fb4f3bebb239f376a7326d05

vhf commented on free-programming-books-ro.md:
```diff
@@ -21,7 +21,7 @@
 ### HMTL
 
-* [HTML](http://tutorialehtml.com/htmlt)
+* [HTML](http://tutorialehtml.com/en/html-tutorial-complete-html-guide/)
```
Should be `http://tutorialehtml.com/ro/introducere-in-html/` for the Romanian version.

davisonio replied:
Done in https://github.com/davisonio/awesome-free-programming-books/commit/226a49895d5f897b84023e684ca2d34c3894e970

vhf commented on free-programming-books-zh.md:
```diff
@@ -190,9 +190,9 @@
 ### NoSQL
 
-* [NoSQL数据库笔谈](http://www.yankay.com/wp-content/NoSql_Database_Note.html) ([PDF](http://yankaycom-wordpress.stor.sinaapp.com/uploads/2012/12/NoSQL%E6%95%B0%E6%8D%AE%E5%BA%93%E7%AC%94%E8%B0%88v2.pdf))
+* [NoSQL数据库笔谈](http://yankaycom-wordpress.stor.sinaapp.com/NoSql_Database_Note.html?q=/wp-content/NoSql_Database_Note.html) ([PDF](http://yankaycom-wordpress.stor.sinaapp.com/uploads/2012/12/NoSQL%E6%95%B0%E6%8D%AE%E5%BA%93%E7%AC%94%E8%B0%88v2.pdf))
```
I get a 403 here.

davisonio replied:
Hmm so do I, I think I'll remove it: https://github.com/davisonio/awesome-free-programming-books/commit/48f405ead77a4aa0a73996cfb98d537d64013499

vhf commented on free-programming-interactive-tutorials-en.md:
```diff
@@ -146,7 +146,7 @@
 
 * [MongoDB Koans](https://github.com/chicagoruby/MongoDB_Koans)
-* [Try MongoDB](http://try.mongodb.org)
+* [Try MongoDB](https://university.mongodb.com/)
```
Not the same as the old one. I suggest we delete this link.

davisonio replied:
Yep, I've removed this in https://github.com/davisonio/awesome-free-programming-books/commit/b61aef2ef82528338963ace92c0f53f00265d58c

vhf commented:
Thanks for all this great work! If you did by hand, awesome! If you scripted it, also awesome, please share your script somewhere! :)

I'm going to review this carefully. I will comment directly on the commit for some specific points, and here for more general points:

- Please don't add a trailing slash to root URIs: we prefer `https://example.com` over `https://example.com/`
- Please don't use overly specific URIs, even if it means the visitor will be redirected. It will prevent some links from breaking in a near future. For example, you added `/details` to each gitbook URIs. Gitbook changing these `/details` URIs to something else is more probable than changing the whole book address. This is only an example, see my comment about `index2.html` directly in the commit.

If you don't have time to address these issues, it's not a problem, I can pull your branch and fix them before committing. Just tell me if you'd rather not take care of this, this way we don't duplicate our efforts!

davisonio commented:
> Thanks for all this great work! If you did by hand, awesome! If you scripted it, also awesome, please share your script somewhere! :)
> 
> I'm going to review this carefully. I will comment directly on the commit for some specific points, and here for more general points:
> 
> Please don't add a trailing slash to root URIs: we prefer https://example.com over https://example.com/
> Please don't use overly specific URIs, even if it means the visitor will be redirected. It will prevent some links from breaking in a near future. For example, you added /details to each gitbook URIs. Gitbook changing these /details URIs to something else is more probable than changing the whole book address. This is only an example, see my comment about index2.html directly in the commit.
> If you don't have time to address these issues, it's not a problem, I can pull your branch and fix them before committing. Just tell me if you'd rather not take care of this, this way we don't duplicate our efforts!

Thanks! It's a pleasure to contribute to such a great resource. This is a bit of scripted and a bit of manual work - my script is based off a very early version of [frankenstein](https://github.com/dkhamsing/frankenstein) but the project is now probably better than mine.

Yep, I agree it doesn't make the URLs look nice - I've removed the trailing slashes in https://github.com/davisonio/awesome-free-programming-books/commit/1169822e5a1b933e6d2a2bfef8f7cd5a2694fc90

I've changed the gitbook links to now go directly to the content (as before) and done a few other things.
